app/workers/worker_base.rb

```diff
@@ -0,0 +1,7 @@
+class WorkerBase
```

Comment:
the advantages i see are:

1. only use one mechanism for sharing behaviour: either inheritance or mixins, and not both
2. make fewer changes to code
3. add behaviour where similar behaviour already exists

ruby allows a lot of ways to achieve the same outcome, and leaves a lot up to the programmer's choice. i'm not sure if i would generalise anything in such a language. i look at monkey patching as a language feature, which is extensively used in [rails active support core extensions](https://github.com/rails/rails/tree/master/activesupport/lib/active_support/core_ext).

i'm happy with the `SidekiqWorker` example you've illustrated, if we want to avoid opening `Sidekiq::Worker`. only suggestion is to call it something like: `Whitehall::Sidekiq::Worker` to make it more explicit, and not easy to confuse with `Sidekiq::Worker`.